pytorch_lightning/trainer/lr_finder.py

```diff
@@ -0,0 +1,288 @@
+"""
+description
+"""
+from typing import Optional
+from abc import ABC
+from pytorch_lightning.core.lightning import LightningModule
+from torch.utils.data import DataLoader
+from tqdm.auto import tqdm
+from pytorch_lightning.callbacks import Callback
+from torch.optim.lr_scheduler import _LRScheduler
+import numpy as np
+import torch
+
+
+class TrainerLRFinderMixin(ABC):
+    def find_lr(self,
```

Comment:
This is awesome! I'd prefer to put this into an argument in the trainer... not have an additional method the user has to call.

@justusschock @ethanwharris thoughts?